src/language/code_generator.py

```diff
@@ -98,6 +101,14 @@ def __new__(cls, base_dir, clang_format=None):
 
         return self
 
+    @property
+    def within_ci(self):
+        """Return True if code is running in a continuous-integration environment, else otherwise"""
+        for varname in ["CI", "CI_PIPELINE_ID", "AZURE_ENV_NAME"]:
+            if varname in os.environ:
+                return True
+        return False
+
```

Comment:
wondering why we need it i.e. what happens if the files are formatted in CI as well? 🤔

Reply:
I tried to understand this but couldn't quite figure it out from looking at our current CMake files and CI scripts...